### PR TITLE
CI: Use new yarn cache in `setup-node` action

### PR DESCRIPTION
**Pull request by oscard0m**
## Description

Add `cache` to workflows using `actions/setup-node`

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@ on:
   push:
     branches:
       - master
-      - 'v*'
+      - "v*"
   pull_request: {}
   schedule:
-    - cron:  '0 3 * * *' # daily, at 3am
+    - cron: "0 3 * * *" # daily, at 3am
```
Comment (Turbo87) [inline .github/workflows/ci.yml:10]: these changes seem unrelated

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -19,6 +19,7 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: 12.x
+          cache: npm
```
Comment (Turbo87) [inline .github/workflows/ci.yml:22]: we're using `yarn` in this repo

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -19,6 +19,7 @@ jobs:
...
 
       - name: get yarn cache dir
```
Comment (Turbo87) [inline .github/workflows/ci.yml:24]: as you can see below we already have caching set up here, so if we migrate to the new way of caching things we should remove the old one

**Review by Turbo87 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (oscard0m) [inline .github/workflows/ci.yml:22]:
```suggestion
          cache: yarn
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -4,10 +4,10 @@ on:
   push:
     branches:
       - master
-      - 'v*'
+      - "v*"
```
Comment (oscard0m) [inline .github/workflows/ci.yml:7]:
```suggestion
      - 'v*'
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (oscard0m) [inline .github/workflows/ci.yml:10]:
```suggestion
    - cron: '0 3 * * *' # daily, at 3am
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ name: Release
 on:
   push:
     tags:
-      - '*'
+      - "*"
```
Comment (oscard0m) [inline .github/workflows/release.yml:6]:
```suggestion
      - '*'
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -15,7 +15,8 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: 12.x
-          registry-url: 'https://registry.npmjs.org'
+          registry-url: "https://registry.npmjs.org"
```
Comment (oscard0m) [inline .github/workflows/release.yml:18]:
```suggestion
          registry-url: 'https://registry.npmjs.org'
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -51,6 +52,7 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node-version }}
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/ci.yml:55]:
```suggestion
          cache: yarn
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -85,6 +87,7 @@ jobs:
       - uses: actions/setup-node@v2
         with:
           node-version: 12.x
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/ci.yml:90]:
```suggestion
          cache: yarn
```

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -16,6 +16,7 @@ jobs:
         with:
           node-version: 12.x
           registry-url: 'https://registry.npmjs.org'
+          cache: npm
```
Comment (Turbo87) [inline .github/workflows/release.yml:19]: I guess it's not necessary here since we're not running `npm install`?

Comment (oscard0m) [reply .github/workflows/release.yml:19]:
```suggestion
```
